### ISSUE TITLE
PHP 8.2.0 Windows release

### ISSUE THREAD

**Issue by tmb-github** [Imagick/imagick#575]
...and we need a corresponding Imagick release. Is one in the works? Just checking.

**Comment by Danack** [comment]
PHP 8.3.0 is also released....however I'm not aware of any BC breaks that require a new version of Imagick.

**Comment by tmb-github** [comment]
@Danack Not true...PHP 8.2.0 was released just a few days ago on 21 December 2022. PHP 8.3.0 is at least a year off. See https://www.php.net/

When I run the new PHP in XAMPP for Windows, I receive the same kind of error I received a year ago at this time when attempting to use Imagick with PHP 8.2.0 before the new version of Imagick was compiled. The error is:

> [22-Dec-2022 00:10:13 UTC] PHP Warning:  PHP Startup: imagick: Unable to initialize module
> Module compiled with module API=20210902
> PHP    compiled with module API=20220829
> These options need to match
>  in Unknown on line 0

Imagick needs to be compiled against the new PHP, apparently. Please advise, and many thanks for your help.

**Comment by Danack** [comment]
Sorry, I appear to have had a brainfart.

I don't actually make the builds for Imagick on Windows. The code does work against PHP 8.2.0 (and against master which will eventually be PHP 8.3.0) 


As per https://github.com/Imagick/imagick/issues/573#issuecomment-1353609195 apparently the box that used to make the windows builds is dead, and hasn't been replaced. I will attempt to chase people who might be working on it.

**Comment by whiteCosmic** [comment]
@Danack @Danack 
I get the same error, using PHP 8.2.0 in XAMPP for Windows.
Looked for imagick version that is compatible with php 8.2.
on [peck.php.net](http://peck.php.net/) I could only find imagick 3.7.0 which is compatible to php 7.4, 8.0 and 8.1.

Can you please let me know if and how can I get the compatible version for php 8.2?

**Comment by williamdes** [comment]
```
/tmp/autopkgtest-lxc.0guk5e6e/downtmp/build.IKf/src/imagick-3.7.0/tests/243_Tutorial_svgExample_basic.phpt
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
-Ok
+Fatal error: Uncaught ImagickException: no decode delegate for this 
image format `SVG' @ error/blob.c/BlobToImage/363 in Standard input code:43
+Stack trace:
+#0 Standard input code(43): Imagick->readImageBlob()
+#1 Standard input code(49): svgExample()
+#2 {main}
+  thrown in Standard input code on line 43
 ```
This is probably also related to php 8.2 
https://bugs.debian.org/1028976

**Comment by Danack** [comment]
@williamdes Nope. This issue is about the box at windows.php.net not having up-to-date builds. I have never provided builds for Windows.

The code (afaik) works fine on PHP 8.2. 

> +Fatal error: Uncaught ImagickException: no decode delegate for this 
> image format `SVG' @ error/blob.c/BlobToImage/363 in Standard input code:43

For whatever reason, that system was not compiled with SVG support.

**Comment by williamdes** [comment]
Thank you for this feedback, I will forward it

**Comment by fedot199614** [comment]
> @Danack Not true...PHP 8.2.0 was released just a few days ago on 8 December 2022. PHP 8.3.0 is at least a year off. See https://www.php.net/
> 
> When I run the new PHP in XAMPP for Windows, I receive the same kind of error I received a year ago at this time when attempting to use Imagick with PHP 8.2.0 before the new version of Imagick was compiled. The error is:
> 
> > [22-Dec-2022 00:10:13 UTC] PHP Warning:  PHP Startup: imagick: Unable to initialize module
> > Module compiled with module API=20210902
> > PHP    compiled with module API=20220829
> > These options need to match
> > in Unknown on line 0
> 
> Imagick needs to be compiled against the new PHP, apparently. Please advise, and many thanks for your help.

the same issue, any updates for this?

**Comment by kamil-karkus** [comment]
I'm not sure if I did something wrong but on 8.2 and latest commit I've got this warning:
```
PHP Warning:  Imagick::__toString() implemented without string return type in Unknown on line 0
```

**Comment by ArthurPatriot** [comment]
https://github.com/Imagick/imagick/issues/573#issuecomment-1431773928

**Comment by Rentner2** [comment]
> > @Danack Not true...PHP 8.2.0 was released just a few days ago on 8 December 2022. PHP 8.3.0 is at least a year off. See https://www.php.net/
> > When I run the new PHP in XAMPP for Windows, I receive the same kind of error I received a year ago at this time when attempting to use Imagick with PHP 8.2.0 before the new version of Imagick was compiled. The error is:
> > > [22-Dec-2022 00:10:13 UTC] PHP Warning:  PHP Startup: imagick: Unable to initialize module
> > > Module compiled with module API=20210902
> > > PHP    compiled with module API=20220829
> > > These options need to match
> > > in Unknown on line 0
> > 
> > 
> > Imagick needs to be compiled against the new PHP, apparently. Please advise, and many thanks for your help.
> 
> the same issue, any updates for this?

me too

**Comment by jamesjohnmcguire** [comment]
Perhaps this needs to be a separate issue, but are there any specific build / make scripts checked in, that people can use to roll their own, if necessary?

**Comment by ArthurPatriot** [comment]
> [#573 (comment)](https://github.com/Imagick/imagick/issues/573#issuecomment-1431773928)

@jamesjohnmcguire

**Comment by christoferd** [comment]
I still cannot find it for PHP 8.2 Windows :-(

**Comment by ArthurPatriot** [comment]
Last version here - https://phpext.phptools.online/extension/images/imagick-121

**Comment by bukka** [comment]
This should be all addressed in the new official Windows builds available at https://github.com/Imagick/imagick/releases/tag/3.8.0
